libs/wire-api/src/Wire/API/MLS/GroupId.hs

```diff
@@ -0,0 +1,96 @@
+-- This file is part of the Wire Server implementation.
+--
+-- Copyright (C) 2022 Wire Swiss GmbH <opensource@wire.com>
+--
+-- This program is free software: you can redistribute it and/or modify it under
+-- the terms of the GNU Affero General Public License as published by the Free
+-- Software Foundation, either version 3 of the License, or (at your option) any
+-- later version.
+--
+-- This program is distributed in the hope that it will be useful, but WITHOUT
+-- ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+-- FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+-- details.
+--
+-- You should have received a copy of the GNU Affero General Public License along
+-- with this program. If not, see <https://www.gnu.org/licenses/>.
+
+module Wire.API.MLS.GroupId
+  ( GroupId (..),
+    parseConvId,
+
+    -- * Convenience utilities
+    convIdToGroupId,
+    decodeConvId,
+  )
+where
+
+import Control.Error.Util
+import qualified Data.Aeson as A
+import qualified Data.Aeson.Types as A
+import qualified Data.ByteString.Lazy as LBS
+import Data.Domain
+import Data.Either.Combinators
+import Data.Id
+import Data.Json.Util (Base64ByteString (Base64ByteString, fromBase64ByteString), base64Schema)
+import Data.Qualified
+import Data.Schema
+import qualified Data.Swagger as S
+import qualified Data.Text.Encoding as T
+import Data.UUID
+import Imports hiding (drop, length, take)
+import Wire.API.Arbitrary
+
+-- | An MLS group ID. It is represented as a byte sequence.
+newtype GroupId = GroupId {unGroupId :: LBS.ByteString}
+  deriving (Eq, Show, Generic)
+  deriving (A.ToJSON, A.FromJSON, S.ToSchema) via (Schema GroupId)
+
+instance Arbitrary GroupId where
+  arbitrary = fmap convIdToGroupId $ Qualified <$> arbitrary <*> mlsDomain
+
+instance ToSchema GroupId where
+  schema =
+    GroupId
+      <$> unGroupId
+      .= named "GroupId" (Base64ByteString .= fmap fromBase64ByteString base64Schema)
+
+-- | Parse a 'GroupId' byte sequence in the network order as a qualified
+-- conversation ID.
+parseConvId :: GroupId -> A.Parser (Qualified ConvId)
```

Comment:
Could we not use `binary` here too for consistency with the rest of the MLS code?